### PR TITLE
Add team name

### PR DESCRIPTION
**Pull request by josephgarner**
Adding team name to hack details and edit button

- refactored hack details and edit hack in hopes of reducing component size
- Added field in edit component to take team name
- Added check in hack details to check for team name and display if true

### CODE REVIEW

File: src/components/edit-hack.js [hunk 1]
```diff
@@ -40,30 +40,30 @@ const useStyles = makeStyles(theme => ({
...
-    dispatch(editingHackIdea(updateData,hackDetails._id))
-    dispatch(clearingHackDetails())
-    history.goBack()
+    dispatch(editingHackIdea(updateData, hackDetails._id));
+    dispatch(clearingHackDetails());
+    history.goBack();
   };
 
   if (hackDetails) {
+    console.log(updateData)
```
Comment (jpeyper) [inline src/components/edit-hack.js]: pretty sure you dont want to keep this :)

Comment (josephgarner) [reply src/components/edit-hack.js]: I mean I might want the user to see all that.
I has been removed

File: src/components/hack-details.js [hunk 2]
```diff
@@ -102,14 +102,9 @@ export function HackDetails({ match, dispatch, userId, hackDetails }) {
               <Grid item xs={10}>
                 <Typography variant="body1">{hackDetails.goal}</Typography>
               </Grid>
-              {hackDetails.team !== undefined ? (
-                <Grid item xs={10} className={classes.marginTop}>
-                  <Typography variant="h6">Team Members:</Typography>
-                </Grid>
+              {hackDetails.team.length !== 0 ? (
+                <Team team={hackDetails.team} name={hackDetails.teamName}/>
               ) : null}
```
Comment (jpeyper) [inline src/components/hack-details.js]: I'd let the `<Team>` component handle the empty state so we can do something like:
```
<p>Looks like no one is in this team yet :(<br />Want to be the first? <a onClick={joinTeam}>Join the team!</a></p>
```

File: src/components/edit-hack.js [hunk 3]
```diff
@@ -40,30 +40,30 @@ const useStyles = makeStyles(theme => ({
...
   };
 
   const onSubmitUpdate = e => {
     e.preventDefault();
-    dispatch(editingHackIdea(updateData,hackDetails._id))
-    dispatch(clearingHackDetails())
-    history.goBack()
+    dispatch(editingHackIdea(updateData, hackDetails._id));
+    dispatch(clearingHackDetails());
+    history.goBack();
```
[the remark this answers is not in the document]
Comment (josephgarner) [reply src/components/edit-hack.js]: Fair point, I didn't think of those situations

File: src/components/subcomponents/hack-team-edit.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      </Grid>
+      <Grid item xs={10} className={classes.marginTop}>
+        <FormControl fullWidth variant="outlined">
+          <OutlinedInput
+            onChange={props.handleOnChange}
+            placeholder="Team Name"
+            name="teamName"
+            multiline
+            rows="1"
+            defaultValue={props.namen}
```
[the remark this answers is not in the document]
Comment (josephgarner) [reply src/components/subcomponents/hack-team-edit.js]: Hot new way to spell name

**Comment by jpeyper** [comment]
you should rebase this onto master (or just merge master) so it runs the tests on the PR.

**Pull request merged**